wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/repeater/data/table/HeadersToolbar.java

```diff
@@ -71,7 +71,10 @@
 
 				for (IColumn<T, S> column : table.getColumns())
 				{
-					columnsModels.add(Model.of(column));
+					if (column.isVisible())
```

Comment:
I'm not sure this is the only involved toolbar that needs to be fixed... Did you run

`mvn clean install`

?